app/controllers/course/discussion/posts_controller.rb

```diff
@@ -0,0 +1,96 @@
+class Course::Discussion::PostsController < Course::ComponentController
+  before_action :set_topic
+  authorize_resource :topic
+  before_action :load_and_authorize_posts, except: [:create, :reply]
```

Comment:
I would lean towards the abstract topic, so that the discussion post controller handles discussion topics only.